### PR TITLE
Fix locale data and upgrade intl-* libs

### PR DESCRIPTION
**Pull request by ericf**
This fixes the locale data by upgrading all the `intl-*` libs to latest, adds support for `selectordinal` arguments in messages, and the `now` option to `{{formatRelative}}`.

## Fixed Locale Data

This vastly improves the locale data in the following ways:

* Added locale data for all leaf locales which differ from their root locale's data; e.g. "zh-Hant-TW" is not the same as "zh".

* Properly de-duplicate data for all CLDR locales by correctly traversing a locale's hierarchy of ancestor locales.

* Added data for the following languages:
  aa, agq, bas, bh, ckb, dav, dje, dsb, dua, dv, dyo, ebu, ewo, guw, guz, hsb, ia, in, iu, iw, jbo, ji, jv, jw, kaj, kam, kcg, khq, ki, kln, kok, ksf, ku, lb, lu, luo, luy, mer, mfe, mgh, mo, mua, nah, nmg, no, nqo, nus, ny, pap, prg, qu, rn, rw, sbp, sh, sma, smi, smj, smn, sms, swc, syr, tk, tl, twq, vai, wa, wo, yav, yi, zgh

## Upgraded `intl-*` Dependencies

The following low-level FormatJS packages have also been upgraded which
provide and take advantaged of the improved CLDR locale data:

* `grunt-extract-cldr-data@2.1.1`
* `intl-messageformat@1.1.0`
* `intl-relativeformat@1.1.0`

## Added `selectordinal` Support for Messages

`intl-messageformat@1.1.0` adds support for `selectordinal` arguments in messages; e.g.:

```
This is my {year, selectordinal,
    one {#st}
    two {#nd}
    few {#rd}
    other{#th}
} birthday.
```

## Added `now` Option to `{{formatRelative}}` Helper

`intl-relativeformat@1.1.0` adds support for specifying the "now" value from which relative time is computed. This change adds a new `now` hash option to the `{{formatRelative}}` helper which will pass along that value to `IntlRelativeFormat#format()` method.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by caridy** [comment]
LGTM

File: src/helpers.js [hunk 1]
```diff
@@ -105,27 +105,72 @@ function registerWith(Handlebars) {
...
+        }
+
+        var locales       = options.data.intl && options.data.intl.locales;
+        var formatOptions = getFormatOptions('relative', format, options);
+        var now           = options.hash.now;
+
+        // Remove `now` from the options passed to the `IntlRelativeFormat`
+        // constructor, because it's only used when calling `format()`.
+        if (formatOptions.hasOwnProperty('now')) {
+            delete formatOptions.now;
```
Comment (caridy) [inline src/helpers.js]: this mutation could happens for every call right?

Comment (ericf) [inline src/helpers.js]: Yeah true, I'll remove the check.

**Pull request merged**